persistence/player_repository_test.go

```diff
@@ -0,0 +1,257 @@
+package persistence
+
+import (
+	"context"
+
+	"github.com/deluan/rest"
+	"github.com/navidrome/navidrome/db"
+	"github.com/navidrome/navidrome/log"
+	"github.com/navidrome/navidrome/model"
+	"github.com/navidrome/navidrome/model/request"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("PlayerRepository", func() {
+	var adminRepo model.PlayerRepository
+	var database *dbxBuilder
+
+	var (
+		adminPlayer1  = model.Player{ID: "1", Name: "NavidromeUI [Firefox/Linux]", UserAgent: "Firefox/Linux", UserId: adminUser.ID, Username: adminUser.UserName, Client: "NavidromeUI", IP: "127.0.0.1", ReportRealPath: true, ScrobbleEnabled: true}
+		adminPlayer2  = model.Player{ID: "2", Name: "GenericClient [Chrome/Windows]", IP: "192.168.0.5", UserAgent: "Chrome/Windows", UserId: adminUser.ID, Username: adminUser.UserName, Client: "GenericClient", MaxBitRate: 128}
+		regularPlayer = model.Player{ID: "3", Name: "NavidromeUI [Safari/macOS]", UserAgent: "Safari/macOS", UserId: regularUser.ID, Username: regularUser.UserName, Client: "NavidromeUI", ReportRealPath: true, ScrobbleEnabled: false}
+
+		players = model.Players{adminPlayer1, adminPlayer2, regularPlayer}
+	)
+
+	BeforeEach(func() {
+		ctx := log.NewContext(context.TODO())
+		ctx = request.WithUser(ctx, adminUser)
+
+		database = NewDBXBuilder(db.Db())
+		adminRepo = NewPlayerRepository(ctx, database)
+
+		for idx := range players {
+			err := adminRepo.Put(&players[idx])
+			Expect(err).To(BeNil())
+		}
+
+		_, err := database.NewQuery("PRAGMA foreign_keys = ON").Execute()
```

Comment:
I thought it was already on for all persistence tests... I think it is better to enable it at the package level, and not just for this test. You could change https://github.com/navidrome/navidrome/blob/55bff343cdaad1f04496f724eda4b55d422d7f17/persistence/persistence_suite_test.go#L24
to:
```
	conf.Server.DbPath = "file::memory:?cache=shared&_foreign_keys=on"
```

Reply:
Yeah, I wasn't sure why it wasn't enabled across the board (e.g. if it was a test setup choice). Fixed!